packages/svelte/src/internal/client/dev/elements.js

```diff
@@ -12,7 +13,7 @@ export function add_locations(fn, filename, locations) {
 	return (/** @type {any[]} */ ...args) => {
 		const dom = fn(...args);
 
-		var node = hydrating ? dom : dom.nodeType === 11 ? dom.firstChild : dom;
+		var node = hydrating ? dom : dom.nodeType === 11 ? get_first_child(dom) : dom;
```

Comment:
are these changes beneficial too? they seem unrelated to the `child` implementation change